packages/core/backend/sirius-components-collaborative/src/main/java/org/eclipse/sirius/components/collaborative/representations/RepresentationEventProcessorChangeListener.java

```diff
@@ -0,0 +1,204 @@
+import org.eclipse.sirius.components.collaborative.api.Monitoring;
+import org.eclipse.sirius.components.collaborative.dto.RepresentationRenamedEventPayload;
+import org.eclipse.sirius.components.collaborative.editingcontext.RepresentationEventProcessorEntry;
+import org.eclipse.sirius.components.collaborative.representations.api.IRepresentationEventProcessorRegistry;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IEditingContextPersistenceService;
+import org.eclipse.sirius.components.core.api.IInput;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Sinks;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Used to react to onChange events from event processors.
+ *
+ * @author mcharfadi
+ */
+@Service
+public class RepresentationEventProcessorChangeListener implements IRepresentationEventProcessorChangeListener {
+
+    public static final String REPRESENTATION_ID = "representationId";
+
+    public static final String REPRESENTATION_LABEL = "representationLabel";
+
+    private static final String LOG_TIMING_FORMAT = "%1$6s";
+
+    private final IRepresentationEventProcessorRegistry representationEventProcessorRegistry;
+
+    private final MeterRegistry meterRegistry;
+
+    private final Logger logger = LoggerFactory.getLogger(RepresentationEventProcessorChangeListener.class);
+
+    private final IDanglingRepresentationDeletionService danglingRepresentationDeletionService;
+
+    private final IEditingContextPersistenceService editingContextPersistenceService;
+
+    public RepresentationEventProcessorChangeListener(IRepresentationEventProcessorRegistry representationEventProcessorRegistry, MeterRegistry meterRegistry,
+                                                      IDanglingRepresentationDeletionService danglingRepresentationDeletionService,
+                                                      IEditingContextPersistenceService editingContextPersistenceService) {
+        this.representationEventProcessorRegistry = Objects.requireNonNull(representationEventProcessorRegistry);
+        this.meterRegistry = Objects.requireNonNull(meterRegistry);
+        this.danglingRepresentationDeletionService = Objects.requireNonNull(danglingRepresentationDeletionService);
+        this.editingContextPersistenceService = Objects.requireNonNull(editingContextPersistenceService);
+    }
+
+    @SuppressWarnings("checkstyle:IllegalCatch")
+    @Override
+    public void onChange(ExecutorService executorService, Sinks.Many<IPayload> payloadSink, Sinks.Many<Boolean> canBeDisposedSink, Sinks.Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, ChangeDescription changeDescription) {
```

Comment:
Unless I missed something, some parameters are not even used (executor service, change description sink). If we can lower some of this coupling it would be great. I'll look into it